examples/tokenvm/controller/action_indexer.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package controller
+
+import (
+	"context"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/examples/tokenvm/actions"
+	"github.com/ava-labs/hypersdk/extension/indexer"
+)
+
+var _ indexer.SuccessfulTxSubscriber = (*actionHandler)(nil)
+
+type actionHandler struct {
+	c *Controller
+}
+
+func (a *actionHandler) Accepted(_ context.Context, tx *chain.Transaction, result *chain.Result) error {
+	for i, act := range tx.Actions {
+		switch action := act.(type) {
+		case *actions.CreateAsset:
+			a.c.metrics.createAsset.Inc()
+		case *actions.MintAsset:
+			a.c.metrics.mintAsset.Inc()
+		case *actions.BurnAsset:
+			a.c.metrics.burnAsset.Inc()
+		case *actions.Transfer:
+			a.c.metrics.transfer.Inc()
+		case *actions.CreateOrder:
+			a.c.metrics.createOrder.Inc()
+			a.c.orderBook.Add(chain.CreateActionID(tx.ID(), uint8(i)), tx.Auth.Actor(), action)
+		case *actions.FillOrder:
+			a.c.metrics.fillOrder.Inc()
+			outputs := result.Outputs[i]
```

Comment:
Wondering if the order-book updates should happen in their own subscriber since they are logically decoupled from the metrics

Reply:
I think it's fine to do more than one thing here.

Doing this async (as it's handled prior to this PR as well) does introduce a race condition because the in-memory order book is not updated during block accept. We may want to support a `SyncAcceptedSubscriber` and `AsyncAcceptedSubscriber` to fix that.

Reply:
Yeah I think long term we're gonna try to get rid of the metrics in here and push it up into hypersdk anyways so... not too concerned